src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -39,6 +66,36 @@ public EnumGroupAndName(string group, string name)
         /// <summary>
         /// Gets the name.
         /// </summary>
-        public string Name { get; }
+        public string Name
+        {
+            get
+            {
+                return _name();
+            }
+        }
+
+        // Equals and GetHashCode must be overloaded to accommodate the _nameFunc
+        public override bool Equals(object obj)
+        {
+            if (!(obj is EnumGroupAndName))
+            {
+                return false;
+            }
+
+            var second = (EnumGroupAndName)obj;
+
+            return string.Equals(Group, second.Group, StringComparison.Ordinal)
+                && _name == second._name;
```

Comment:
Avoiding overriding `Equals` and `GetHashCode` would be my preference, but I thought you had said we needed to because the default implementation would be wrong. I don't know of anywhere that will use it so I'll happily remove these two.